File: CodeRun_Boost/7264_Bridge_to_the_palace/solution.dart
import 'dart:math';

int solve(int n, List<int> a) {
  a.sort();

  int left = 0;
  int maxKeep = 0;

  for (int right = 0; right < n; right++) {
    while (a[right] - a[left] >= n) {
      left++;
    }

    int currentKeep = right - left + 1;

    if (currentKeep > maxKeep) {
      maxKeep = currentKeep;
    }
  }

  int minMoves = n - maxKeep;

  return minMoves;
}

File: CodeRun_Boost/7264_Bridge_to_the_palace/solution.swift
import Foundation

public func solve(n: Int, a: inout [Int]) -> Int {
    a.sort()

    var left = 0
    var maxKeep = 0

    for right in 0..<n {
        while a[right] - a[left] >= n {
            left += 1
        }

        let currentKeep = right - left + 1
        if currentKeep > maxKeep {
            maxKeep = currentKeep
        }
    }

    let minMoves = n - maxKeep
    return minMoves
}